### PR TITLE
Python-version-specific install path for chpl-venv (chpldoc works)

### PR DESCRIPTION
**Pull request by awallace-cray**
Python-version-specific install path for chpl-venv

Same as PR #4369 except now getChplPythonVersion() (in files.cpp) strips the newline

### CODE REVIEW

**Comment by ronawho** [comment]
This seems backwards to me. Instead of trying to trim newlines in the compiler, why don't you just stop printing the newline in chpl_python_version.py?

**Comment by ronawho** [comment]
Or if you want it to print a newline by default, you can add a `--no-newline` option to the script and throw that in the compiler

**Comment by awallace-cray** [comment]
I only want to touch the one thing that didn't work.

**Comment by ronawho** [comment]
That doesn't make sense to me, it didn't work because chpl_python_version.py prints out a newline. Why would you go through the effort or removing a newline instead of just not printing it out in the first place.

This would be cleaner if chpl_python_version.py didn't print a newline, or at the very least if your moved the newline removal code into getChplPythonVersion()

File: compiler/util/files.cpp [hunk 1]
```diff
@@ -440,6 +440,18 @@ std::string runPrintChplEnv(std::map<std::string, const char*> varMap) {
...
 }
 
+std::string getChplPythonVersion() {
+  // Runs util/chplenv/chpl_python_version.py and left-trims the string
+
+  std::string command = "";
+  command += std::string(CHPL_HOME) + "/util/chplenv/chpl_python_version.py 2> /dev/null";
+
+  std::string pyVer = runCommand(command);
+  pyVer.erase( pyVer.find_last_not_of(" \n\r\t")+1).c_str();
```
Comment (ronawho) [inline compiler/util/files.cpp]: remove `.c_str()`

Also, why are you stripping spaces and tabs too?

**Comment by ronawho** [comment]
This PR message is not very helpful. It forces you to look at the old PR, just to figure out what this one is doing, and forces you to diff them to figure out what actually changed.

It should just say what it does, and maybe at the end say "Same as 4369, but strips newline in the getChplPythonVersion"

**Comment by awallace-cray** [comment]
This probably needs re-tested now, because PR #4416

**Comment by awallace-cray** [comment]
Rebased and re-tested. Should be ready now.

**Comment by awallace-cray** [comment]
Ready for review.

File: compiler/util/files.cpp [hunk 2]
```diff
@@ -440,6 +440,18 @@ std::string runPrintChplEnv(std::map<std::string, const char*> varMap) {
   return runCommand(command);
 }
 
+std::string getChplPythonVersion() {
+  // Runs util/chplenv/chpl_python_version.py and left-trims the string
```
Comment (ronawho) [inline compiler/util/files.cpp]: I don't think this is left-triming, wouldn't that be taking characters off the left? I'd just say `Runs util/chplenv/chpl_python_version.py stripping newlines`

File: compiler/util/files.cpp [hunk 3]
```diff
@@ -440,6 +440,18 @@ std::string runPrintChplEnv(std::map<std::string, const char*> varMap) {
...
+
+  std::string command = "";
+  command += std::string(CHPL_HOME) + "/util/chplenv/chpl_python_version.py 2> /dev/null";
+
+  std::string pyVer = runCommand(command);
+  pyVer.erase( pyVer.find_last_not_of("\n\r")+1);
```
Comment (ronawho) [inline compiler/util/files.cpp]: Remove extra space before `pyVer.` if you're going to be editing this file anyways

File: util/chplenv/__init__.py [hunk 4]
```diff
@@ -28,6 +28,7 @@
     'chpl_3p_qthreads_configs',
     'chpl_3p_re2_configs',
     # General purpose helpers
+    'chpl_python_version',
```
Comment (ronawho) [inline util/chplenv/__init__.py]: should be alphabetize within the sublist. i.e. should come after chpl_home_utils.

**Comment by awallace-cray** [comment]
okay. done.

**Comment by ronawho** [comment]
looks good

**Pull request merged**